contracts/token-factory/contracts/ERC677.sol

```diff
@@ -0,0 +1,26 @@
+pragma solidity ^0.5.2;
+
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "openzeppelin-solidity/contracts/utils/Address.sol";
+
+contract ERC677 is ERC20 {
+    event Transfer(address indexed from, address indexed to, uint value, bytes data);
+
+    function transferAndCall(address _to, uint _value, bytes calldata _data) external returns (bool) {
+      require(_to != address(this));
+
+      _transfer(msg.sender, _to, _value);
+
+      emit Transfer(msg.sender, _to, _value, _data);
```

Comment:
I have some concerns that the original event of Transfer is still being sent. 
I imagine that this can mess up apps that listen to Transfer events.. For example the POA explorer.
But we can merge this and see what happens :)